scripts/Get-TagsDocumentation.ps1

```diff
@@ -0,0 +1,31 @@
+param(
+    [string]$Branch='master',
+    [string]$ImageBuilderImageName='microsoft/dotnet-buildtools-prereqs:image-builder-jessie-20171005132855',
+    [string]$RepoName
+)
+
+$ErrorActionPreference = 'Stop'
+$repoRoot = Split-Path -Path "$PSScriptRoot" -Parent
+
+if ([String]::IsNullOrWhiteSpace($RepoName))
+{
+    $remoteUrl = $null
+    if ([Uri]::TryCreate(((git config --get remote.origin.url) | Out-String), [UriKind]::Absolute, [ref]$remoteUrl))
+    {
+        $RepoName = [System.IO.Path]::GetFileNameWithoutExtension($remoteUrl.ToString())
+    }
+}
+
+if ([String]::IsNullOrWhiteSpace($RepoName))
```

Comment:
This should be moved into the previous `if` body as there is no point in checking `$RepoName` twice when the caller specifies one.

Reply:
Good one! It would be -

```PowerShell
if ([String]::IsNullOrWhiteSpace($RepoName))
{
    $remoteUrl = $null
    if ([Uri]::TryCreate(((git config --get remote.origin.url) | Out-String), [UriKind]::Absolute, [ref]$remoteUrl))
    {
        $RepoName = [System.IO.Path]::GetFileNameWithoutExtension($remoteUrl.ToString())
    }
    if ([String]::IsNullOrWhiteSpace($RepoName))
    {
        Write-Error 'Could not automatically determine repository name. Add -RepoName <REPO> to override.'
    }
}
```